packages/pricing/src/repositories/pricing.ts

```diff
@@ -63,25 +63,66 @@ export class PricingRepository
       psma: "price_set_money_amount",
     })
       .select({
-        id: "psma.id",
-        price_set_id: "psma.price_set_id",
-        money_amount_id: "psma.money_amount_id",
-        number_rules: "psma.number_rules",
+        id: "psma1.id",
+        price_set_id: "psma1.price_set_id",
+        money_amount_id: "psma1.money_amount_id",
+        number_rules: "psma1.number_rules",
+        price_list_id: "psma1.price_list_id",
+        pl_number_rules: "pl.number_rules",
+        pl_type: "pl.type",
       })
-      .leftJoin("price_set_money_amount as psma1", "psma1.id", "psma.id")
-      .leftJoin("price_rule as pr", "pr.price_set_money_amount_id", "psma.id")
+      .leftJoin("price_set_money_amount as psma1", "psma1.id", "psma1.id")
+      .leftJoin("price_rule as pr", "pr.price_set_money_amount_id", "psma1.id")
+      .leftJoin("price_list as pl", "pl.id", "psma1.price_list_id")
+      .leftJoin("price_list_rule as plr", "plr.price_list_id", "pl.id")
+      .leftJoin(
+        "price_list_rule_value as plrv",
+        "plrv.price_list_rule_id",
+        "plr.id"
+      )
+      .leftJoin("rule_type as plrt", "plrt.id", "plr.rule_type_id")
       .leftJoin("rule_type as rt", "rt.id", "pr.rule_type_id")
+      .orderBy("pl.number_rules", "desc")
       .orderBy("number_rules", "desc")
-      .orWhere("psma1.number_rules", "=", 0)
-      .groupBy("psma.id")
-      .having(knex.raw("count(DISTINCT rt.rule_attribute) = psma.number_rules"))
-
-    for (const [key, value] of Object.entries(context)) {
-      psmaSubQueryKnex.orWhere({
-        "rt.rule_attribute": key,
-        "pr.value": value,
-      })
-    }
+      .orderBy([
+        { column: "number_rules", order: "desc" },
+        { column: "pl.number_rules", order: "desc" },
+      ])
+      .groupBy("psma1.id", "pl.id")
+      .having(
+        knex.raw(
+          "count(DISTINCT rt.rule_attribute) = psma1.number_rules AND psma1.price_list_id IS NULL"
+        )
+      )
+      .orHaving(
+        knex.raw(
+          "count(DISTINCT plrt.rule_attribute) = pl.number_rules AND psma1.price_list_id IS NOT NULL"
+        )
+      )
```

Comment:
question: should we include date filtering here as well?

Reply:
discussed on chat, resolving this.